core/iwasm/interpreter/wasm.h

```diff
@@ -867,14 +876,59 @@ wasm_dpgo_get_cur_prof_cnt_info(WASMModule *module, uint32 func_idx,
 }
 
 static inline void
-wasm_dpgo_dump_prof_cnt_info(struct WASMProfCntInfo *cnt, char *buf,
-                             uint32 buf_size)
+wasm_dpgo_prof_cnt_info_to_string(struct WASMProfCntInfo *cnt, char *buf,
+                                  uint32 buf_size)
 {
     snprintf(buf, buf_size, "  OP:0x%x,OFFSET:%u,CNT:%u,1ST_IDX:%u",
              cnt->opcode, cnt->offset, cnt->counter_amount,
              cnt->first_counter_idx);
 }
 
+/* search in `prof_cnts_info[i]` with the given `offset` of `struct
+ * WASMProfCntInfo` and return `counter_amount` and `first_counter_idx`
+ */
+static inline bool
```

Comment:
Function body may be a little big, had better not use inline? Will we put the implementation into core/iwasm/compilation later?

Reply:
You are right. I will do that in the later code refractory PR.